nutaudet/Target.cxx

```diff
@@ -375,18 +375,11 @@ void Target::ConstructGeometry()
   TGeoVolume *volBrick = new TGeoVolume("Brick",Brick,air);
   volBrick->SetLineColor(kCyan);
   volBrick->SetTransparency(1);
-
-  TGeoBBox *Absorber; //need to separate the two cases
-  TGeoVolume *volAbsorber;
-
-  if (fDesign < 4){
-   Absorber = new TGeoBBox("Pb", EmulsionX/2, EmulsionY/2, LeadThickness/2);
-   volAbsorber = new TGeoVolume("Lead",Absorber,lead);
-  }
-  else{
-   Absorber = new TGeoBBox("W", EmulsionX/2, EmulsionY/2, LeadThickness/2);
-   volAbsorber = new TGeoVolume("Tungsten",Absorber,tungsten);
-  }
+  //need to separate the two cases, now with a ternary operator
+  TGeoBBox *Absorber = (fDesign < 4) ? 
+    new TGeoBBox("Pb", EmulsionX/2, EmulsionY/2, LeadThickness/2) : new TGeoBBox("W", EmulsionX/2, EmulsionY/2, LeadThickness/2);
+  TGeoVolume *volAbsorber = (fDesign < 4) ?
+    new TGeoVolume("Lead",Absorber,lead) : new TGeoVolume("Tungsten",Absorber,tungsten);
```

Comment:
Do we maybe need `[[maybe_unused]]` for these later, or are they both "used" in a way that's visible to the compiler? Please check whether there is a warning.

You could use `auto` here as well, now that the initialisation determines the type in an unambiguous way.

We could probably simplify this further by only changing the material of `volAbsorber` and keeping everything else the same.

Reply:
Absorber and volAbsorber are always used, since they are part of the main Target. 
The only difference is the material, if it is lead (originally planned for SHiP) or tungsten (the most updated designs). 
I agree for the `auto` comment